app/routers/world_clock.py

```diff
@@ -0,0 +1,744 @@
+                             'Afternoon':
+                             [('13:00:00', '15:59:59'), 'OK'],
+                             'Late afternoon':
+                             [('16:00:00', '16:59:59'),
+                              'Can be considered'],
+                             'Early evening':
+                             [('17:00:00', '18:59:59'),
+                              'Can be considered'],
+                             'Evening':
+                             [('19:00:00', '20:59:59'), 'Better not'],
+                             'Night':
+                             [('21:00:00', '23:59:59'), 'Better not'],
+                             'Late night':
+                             [('00:00:00', '04:59:59'),
+                              'Not possible'], }
+
+
+def normalize_continent_name(continent_name: str) -> Optional[str]:
+    """Normalize a given continent name to the continent name
+       that appears in the timezone list.
+
+
+    Args:
+        continent_name (str): The given continent name.
+
+
+    Returns:
+        str: The normalized continent name.
+    """
+    for continent in CONTINENTS:
+        if continent in continent_name:
+            return continent
+    return continent_name
+
+
+def get_continent(country_name: str) -> Optional[str]:
+    """Get the continent of a given country.
+
+
+    Args:
+        country_name (str): The given country name.
+
+
+    Returns:
+        str: The suitable continent name.
+    """
+    details = requests.get(COUNTRY_TO_CONTINENT_DATA_SET).json()
+    for country_element in details:
+        if country_name.title() in \
+                country_element[COUNTRY_NAME_KEY_IN_CONTITNENTS]:
+            return normalize_continent_name(
+                country_element[CONTINENT_NAME_KEY])
+
+
+def get_country(city_name: str) -> Optional[str]:
+    """Get the country of a city.
+
+
+    Args:
+        city_name (str): The given city name.
+
+
+    Returns:
+        str: The suitable country name.
+    """
+    details = requests.get(CITY_TO_COUNTRY_DATA_SET).json()
+    for city_element in details:
+        if city_name.title() in city_element[CITY_NAME_KEY].title():
+            return city_element[COUNTRY_NAME_KEY_IN_CITIES]
+
+
+def get_subcountry(city_name: str) -> Optional[str]:
+    """Get the subcountry of a city.
+
+
+    Args:
+        city_name (str): The given city name.
+
+
+    Returns:
+        str: The suitable subcountry name.
+    """
+    details = requests.get(CITY_TO_COUNTRY_DATA_SET).json()
+    for city_element in details:
+        if city_name.title() in city_element[CITY_NAME_KEY].title():
+            return city_element[SUBCOUNTRY_NAME_KEY_IN_CITIES]
+
+
+def get_api_data(url: str) -> Optional[List[str]]:
+    """Get the data from an API url.
+
+    Args:
+        url (str):  The API url.
+
+    Returns:
+        list: The data.
+    """
+    try:
+        resp = requests.get(url)
```

Comment:
this is the core of my logic and the pivot for the other functions - its for getting the current time around the world. I really don't know what I can do here...
The API requests outside this function can be dealt by treating them as reading a file, but this is not the case here...

Reply:
You can move it to JS - there are even example of other people who already did it :)

Reply:
Can you please elaborate? That's exactly the part I do not understand :(
1. Can I leave the call to the API?
2. Will some of the code be in Python and some in JS? How does it actually fit in?
3. Alternatively, does it mean that I have to give up all my code and just use JS? I am really afraid from this because I have invested a lot of time and effort in the code and in addition, I am less familiar with JS.

Reply:
I am having a really hard time with making the code asynchronous and the submission time is already pressing so I am very stressed out.
I researched hours online, asked at StackOverflow, but failed to to put it together into a code that works. 
1. Do I need to change all the functions to "async def"? 
	1.1. Do I have to change the tests to async def as a result? 
2. Should the call to the API be with "await"? why? 
	2.1. If so - for some reason I can not refer to the information obtained from the call to the API in other functions for the reason that "'coroutine' object is not iterable python".

As a beginning, I tried to modify the code of the "get_api_data" function according to the documentation to the following function:
def get_api_data (url: str) -> Optional [List [Any]]:
    try:
        async with httpx.AsyncClient () as client:
             resp = await client.get (url)
           return resp.json ()
    except httpx.HTTPError as errh:
      print ("Http Error:", errh)

But it just does not work ...
Can you guide me?